pkg/chunkenc/memchunk.go

```diff
@@ -229,7 +233,9 @@ func NewByteChunk(b []byte) (*MemChunk, error) {
 		// Verify checksums.
 		expCRC := binary.BigEndian.Uint32(b[blk.offset+l:])
 		if expCRC != crc32.Checksum(blk.b, castagnoliTable) {
-			return bc, ErrInvalidChecksum
+			level.Error(logger).Log("msg", "Checksum doesnot match for chunck block")
+			continue
+			//return bc, ErrInvalidChecksum
```

Comment:
You should may be look at who is using this and keep that error returned there. The usage of this should verify if this is a checksum error we can skip.

Reply:
Basically I wouldn't make the change here, but at a higher level.

Reply:
At the block level, should we just skip the whole Chunk if one of the blocks has invalid checksum? or we should continue iterating other blocks?

Reply:
we should continue iterating other blocks